File: Gymify.Services/Database/Membership.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Gymify.Services.Database
{
    public class Membership
    {
        [Key]

        public int Id {get; set;}

        public string Name {get; set;}

        public int MonthlyPrice {get; set;}

        public int YearPrice {get; set;}
        public DateTime CreatedAt {get; set;}
    }
}

File: UI/gymify_desktop/lib/models/membership.dart
import 'package:json_annotation/json_annotation.dart';

part 'membership.g.dart';

@JsonSerializable()
class Membership {
  final int id;
  final String name;
  final double monthlyPrice;
  final double yearPrice;
  final DateTime createdAt;

  Membership({
    required this.id,
    required this.name,
    required this.monthlyPrice,
    required this.yearPrice,
    required this.createdAt,
  });

  factory Membership.fromJson(Map<String, dynamic> json) =>
      _$MembershipFromJson(json);

  Map<String, dynamic> toJson() => _$MembershipToJson(this);
}
